src/Service/SQS.service.js

```diff
@@ -262,9 +283,10 @@ export default class SQSService extends DependencyAwareClass {
    * @param queue          string
    * @param messageObject  object
    * @param messageGroupId string
+   * @param failureMode
```

Comment:
I had added this:

```
   * By default, this method will succeed even if it fails to send to SQS. You
   * can change this behaviour by setting `failureMode` to `throw`.
   *
   * @param {string} queue
   * @param {object} messageObject
   * @param {string | null} messageGroupId
   * @param {'catch' | 'throw'} failureMode Choose how failures are handled:
   *   - `catch`: errors will be caught and logged. This is the default.
   *   - `throw`: errors will be thrown, causing promise to reject.
   *
```

I guess you'd want to change that to say "use values from `SQS_PUBLISH_FAILURE_MODES`", although I don't really like that as much; it's another thing import and it has a long name. Specifying it as a string type also means we get autocompletion!

Reply:
If this was TypeScript I wouldn't have used an object but the union type as you did. I agree with you, I am happy with your change!

Reply:
60a7217d2cb5c5bd8ed0ce8568ec5cc02fe9fd0c

Reply:
Cool, thanks, I wasn't sure if you were going for an `enum` approach! Could you also update the values in `SQS_PUBLISH_FAILURE_MODES` to be lowercase, to match the type?

Reply:
91f9403fc92d8f686f35ff21b813e54d0b7913a0